src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -121,16 +119,61 @@ define(function (require, exports, module) {
 
                                 PreferencesManager.setViewState("UUID",      oneTimeHealthData.uuid);
                                 PreferencesManager.setViewState("OlderUUID", oneTimeHealthData.olderuuid);
-
                                 return result.resolve(oneTimeHealthData);
                             }
                         }
                     });
 
             });
-
         return result.promise();
     }
+    
+    //Get Analytics data
+    
+    function getAnalyticsData(category, subcategory, type, subtype) {
+        var userUuid = PreferencesManager.getViewState("UUID"),
+            olderUuid = PreferencesManager.getViewState("OlderUUID"),
+            isoDate = new Date().toISOString(),
+            userAgent = window.navigator.userAgent,
+            isNeedToSetViewState = false;
+        if (!userUuid) {
+            userUuid = uuid.v4();
```

Comment:
We will end up setting wrong UUID as we are not using the macHash and this function is getting executed before sendHealthData call.

Reply:
Now I am calling it once we have send health data server

Reply:
Changed with current commit

Reply:
the following piece of code is not required in that case - 
``` javascript
if (isNeedToSetViewState) {
            PreferencesManager.setViewState("UUID", userUuid);
            PreferencesManager.setViewState("OlderUUID", olderUuid);
        }
```